bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/widgets/tagbrowser.directive.js

```diff
@@ -0,0 +1,248 @@
+/*
+ * #%L
+ * wcm.io
+ * %%
+ * Copyright (C) 2022 wcm.io
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+(function (angular, Coral, $) {
+  "use strict";
+
+  angular.module("io.wcm.caconfig.widgets")
+    .directive("caconfigTagbrowser", tagbrowser);
+
+  tagbrowser.$inject = ["templateUrlList", "directivePropertyPrefixes", "$timeout", "$rootScope", "configService", "tagbrowserService"];
+
+  function tagbrowser(templateList, directivePropertyPrefixes, $timeout, $rootScope, configService, tagbrowserService) {
+    var directive = {
+      replace: true,
+      templateUrl: templateList.tagbrowser,
+      scope: {
+        property: "="
+      },
+      link: link
+    };
+
+    return directive;
+
+    function link(scope, element) {
+      var multiValue = scope.property.metadata.multivalue;
+
+      if(multiValue) {
+        scope.effectiveValues = [];
+        scope.values = [];
+
+        setValueArray(scope.property.effectiveValue, scope.effectiveValues);
+        setValueArray(scope.property.value, scope.values);
+      }
+
+      var prefix = directivePropertyPrefixes.tagbrowser;
+      var props = scope.property.metadata.properties;
+
+      var tagfieldName = "tags-" + Math.floor(Math.random() * 100000);
+
+      var options = {};
+      var tagfieldWidget;
+      var suggestionOverlay;
+      angular.forEach(props, function (value, prop) {
+        var propName;
+        // if the property starts with the prefix "tagbrowser" followed by a tagbrowser property name
+        // remove the "tagbrowser" prefix and use the remaining part as option name
+        if (prop && prop.length > prefix.length && prop.substring(0, prefix.length) !== -1) {
+          propName = prop.substring(prefix.length);
+          options[propName.charAt(0).toLowerCase() + propName.slice(1)] = props[prop];
+        }
+      });
+
+      // get root path from config or take /content/cq:tags as default
+      options.rootPath = options.rootPath || "/content/cq:tags";
+      // if rootPathContext is set root path to current context path
+      if (options.rootPathContext === "true") {
```

Comment:
i think this option is not used - copy & paste from path browser

Reply:
`options.rootPath` is used for the pickersrc and suggestionsrc in tagbrowser.directive.js.

```JavaScript
tagfieldWidget.setAttribute("pickersrc", tagbrowserService.getPickerSrc(options.rootPath, selectionCount));
//....
tagbrowserService.getSuggestionSrc(options.rootPath));
```

I did however removed line 72 and following since the rootPathContext is also removed from the backend `EditorProperties`, as you asked above.